framework/validators/IpValidator.php

```diff
@@ -0,0 +1,442 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\validators;
+
+use Yii;
+use yii\base\InvalidConfigException;
+
+/**
+ * IpValidator validates that the attribute value is a valid IPv4/IPv6 address or subnet.
+ * May change attribute's value if normalization is enabled.
+ *
+ * @author SilverFire <d.naumenko.a@gmail.com>
+ * @since 2.0.4
+ */
+class IpValidator extends Validator
+{
+    /**
+     * @const integer the length of IPv6 address in bits
+     */
+    const IPV6_ADDRESS_LENGTH = 128;
+
+    /**
+     * @const integer the length of IPv4 address in bits
+     */
+    const IPV4_ADDRESS_LENGTH = 32;
+
+    /**
+     * @const integer is used to change check order by [[isAllowed]]
+     * @see isAllowed
+     */
+    const ORDER_ALLOW_DENY = 0;
+
+    /**
+     * @const integer is used to change check order by [[isAllowed]]
+     * @see isAllowed
+     */
+    const ORDER_DENY_ALLOW = 1;
+
+    /**
+     * @var boolean whether support of IPv6 addresses is enabled
+     */
+    public $ipv6 = true;
+
+    /**
+     * @var boolean whether support of IPv4 addresses is enabled
+     */
+    public $ipv4 = true;
+
+    /**
+     * @var boolean|string whether address can be a CIDR subnet
+     *   boolean - whether an IP address with the CIDR subnet notation will be able to pass the validation
+     *   string  - value 'only' to validate only the addresses with the CIDR
+     */
+    public $subnet = false;
```

Comment:
Suggest using these values:

* `true`: subnet is required
* `false` (default): cannot have subnet
* null: can accept subnet.